crates/context_aware_config/src/api/default_config/handlers.rs

```diff
@@ -28,12 +28,18 @@ use diesel::{
     ExpressionMethods, PgConnection, QueryDsl, RunQueryDsl,
 };
 use jsonschema::{Draft, JSONSchema, ValidationError};
+use regex::Regex;
 use serde_json::{from_value, json, Map, Value};
 use service_utils::{
     result as superposition,
     service::types::{AppState, DbConnection},
 };
 
+// the first part of the regex covers for key names more than 1 character in length and
+// the second part of the regex covers for key names that is 1 character in length
+const KEY_NAME_REGEX: &str =
+    "^([a-zA-Z0-9-_][a-zA-Z0-9-_.]{0,62}[a-zA-Z0-9-_]|[a-zA-Z0-9-_])$";
```

Comment:
`^[a-zA-Z0-9-_]([a-zA-Z0-9-_\.]{0,62}[a-zA-Z0-9-_])?$`
can we try something like this, with this we would not need to have to break the regex into multiple parts without using any look arounds

Reply:
also we would need to do an escape sequence for `.` to `\.` as `.` is the accept any character regex symbol

Reply:
Inside the character class the `.` need not be escaped I believe - it is matched literally.

Reply:
yeah I was not aware of it